File: guilib-api/src/main/java/net/developertobi/guilib/api/gui/GuiProperties.java
package net.developertobi.guilib.api.gui;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface GuiProperties {
    String id();
    int rows() default 1;
    int columns() default 9;
    String permission() default "";
    boolean closeable() default true;
    boolean playSoundOnClick() default true;
    boolean playSoundOnOpen() default true;
    boolean playSoundOnClose() default true;
    boolean playSoundOnPageSwitch() default true;
}

File: guilib-api/bin/main/net/developertobi/inventorylib/api/inventory/GuiProperties.kt
package net.developertobi.inventorylib.api.inventory

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
annotation class GuiProperties(
    val id: String,
    val rows: Int = 1,
    val columns: Int = 9,
    val permission: String = "",
    val closeable: Boolean = true,
    val playSoundOnClick: Boolean = true,
    val playSoundOnOpen: Boolean = true,
    val playSoundOnClose: Boolean = true,
    val playSoundOnPageSwitch: Boolean = true
)

